cmd/yggd/worker.go

```diff
@@ -0,0 +1,262 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"git.sr.ht/~spc/go-log"
+	"github.com/pelletier/go-toml"
+	"github.com/redhatinsights/yggdrasil"
+	"github.com/rjeczalik/notify"
+)
+
+type workerConfig struct {
+	Exec      string   `toml:"exec"`
+	Protocol  string   `toml:"protocol"`
+	Env       []string `toml:"env"`
+	delay     time.Duration
+	directive string
+}
+
+// loadWorkerConfig reads the contents of file and parses it into a workerConfig
+// value.
+func loadWorkerConfig(file string) (*workerConfig, error) {
+	data, err := ioutil.ReadFile(file)
+	if err != nil {
+		return nil, fmt.Errorf("cannot read file: %w", err)
+	}
+
+	var config workerConfig
+	if err := toml.Unmarshal(data, &config); err != nil {
+		return nil, fmt.Errorf("cannot load config: %w", err)
+	}
+	config.directive = strings.TrimSuffix(filepath.Base(file), filepath.Ext(file))
+
+	return &config, nil
+}
+
+// startWorker constructs a command to execute from the given workerConfig,
+// starts it, and starts a goroutine that waits for the process to exit. If not
+// nil, started is invoked after the process is started. Likewise, when the
+// process is stopped, stopped is invoked.
+func startWorker(config workerConfig, started func(pid int), stopped func(pid int)) error {
+	argv := strings.Split(config.Exec, " ")
+
+	program := argv[0]
+	var args []string
+	if len(argv) > 1 {
+		args = argv[1:]
+	}
+
+	env := []string{
+		"PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
+		"YGG_CONFIG_DIR=" + filepath.Join(yggdrasil.SysconfDir, yggdrasil.LongName),
+		"YGG_LOG_LEVEL=" + log.CurrentLevel().String(),
+		"YGG_CLIENT_ID=" + ClientID,
+		"http_proxy=" + os.Getenv("http_proxy"),
+		"HTTP_PROXY=" + os.Getenv("HTTP_PROXY"),
+		"HTTPS_PROXY=" + os.Getenv("HTTPS_PROXY"),
```

Comment:
Is there any reason to not support the lowercase https_proxy env var?

Reply:
I was modeling it off the [curl manpage](https://www.man7.org/linux/man-pages/man1/curl.1.html#ENVIRONMENT), but rereading that, I see that both upper and lower cases are supported, with the lower-case taking precedence. I'll refactor this to behave more like how `curl` is documented.